aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1134,6 +1134,22 @@ static void addmm_impl_cpu_(
     return;
   }
 
+  if (!result._is_zerotensor() and (m1._is_zerotensor() or m2._is_zerotensor())){
```

Comment:
this is incorrect. you need a separate error check. please revert to the old change for the rest

`TORCH_CHECK(!!result._is_zerotensor(), ...);`